sklearn/linear_model/least_angle.py

```diff
@@ -689,6 +689,8 @@ def fit(self, X, y, Xy=None):
                     a[0] for a in (self.alphas_, self.active_, self.coef_path_,
                                    self.coef_)]
                 self.n_iter_ = self.n_iter_[0]
+            else:
+                self.coef_ = np.array(self.coef_)
```

Comment:
it would be  more efficient to preallocate self.coef_ to an empty array rather than first having a list than copying the data to an array